### PR TITLE
URW-7: Navbar Component

### PR DESCRIPTION
**Pull request by corbindonner**
- Link to the ticket [URW-7](https://umnrldev.atlassian.net/browse/URW-7)
- Added 2 pages (Home and About)
- Created Navbar in header component with 4 buttons
- (UMNRL Home, About, Discord, and Twitter)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,25 @@
 import React from "react";
+import Navbar from "./components/Navbar/navbar";
+import About from "./components/pages/about";
+import Home from "./components/pages/home";
+
 
 function App() {
-  return <div className="App">To be implemented</div>;
+  let component;
+  switch (window.location.pathname) {
```
Comment (MDixey17) [inline src/App.tsx]: Let's try something different with navigation. I would like you to explore `react-router-dom` and the React router. This is a modern, secure way to navigate between pages. Some resources:
- [My old website that used routing - code base](https://github.com/MDixey17/personal-website/blob/main/src/App.js)
- [W3Schools](https://www.w3schools.com/react/react_router.asp)

File: src/components/Navbar/navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export default function Navbar() {
```
Comment (MDixey17) [inline src/components/Navbar/navbar.tsx]: Let's use the arrow syntax as this is more modern practice
`export const Navbar = () => {...}`

File: src/components/pages/about.tsx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default function About() {
+    return <h1>About</h1>
```
Comment (MDixey17) [inline src/components/pages/about.tsx:2]: Something minor, let's follow a modern file naming convention:
- `.tsx` files should have words capitalize with no spaces (ex: `Navbar.tsx`, `NavbarElements.tsx`, `About.tsx`)
- `.css` files can be camel-case (ex: `globalStyles.css`, `styles.css`)

File: src/components/pages/about.tsx [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export default function About() {
```
Comment (MDixey17) [inline src/components/pages/about.tsx]: Arrow syntax
`export const About = () => {...}`

File: src/components/pages/home.tsx [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export default function Home() {
```
Comment (MDixey17) [inline src/components/pages/home.tsx]: Arrow syntax
`export const Home = () => {...}`

**Review by MDixey17 — CHANGES_REQUESTED** [review]
Solid start, let me run it locally to see if it needs any other changes

EDIT: looks great locally

**Comment by corbindonner** [comment]
- First implementation of react-router-dom's BrowserRouter
- Changed filenames to conform to naming convention
- Changed function signatures to arrow syntax

File: src/App.tsx (hunk 1 above)
Comment (corbindonner) [reply src/App.tsx]: I didn't know whether or not to use react-router-dom for external sites like twitch, twitter and discord. I left them as anchor tags for now.

Comment (MDixey17) [reply src/App.tsx]: External links like Twitter, Twitch, Discord, etc. should always be anchor tags, so you are good there

File: src/App.tsx [hunk 6]
```diff
@@ -1,7 +1,24 @@
...
 
 function App() {
-  return <div className="App">To be implemented</div>;
+  return (
+    <>
+      <BrowserRouter>
+        <Navbar></Navbar>
+        <Routes>
+          <Route path ="/" element={<Home />} />
+          <Route path="about" element={<About />} />
```
Comment (MDixey17) [inline src/App.tsx:15]: Perfect!

**Review by MDixey17 — APPROVED** [review]
Very nice

**Pull request merged**